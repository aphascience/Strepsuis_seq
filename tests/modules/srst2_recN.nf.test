nextflow_process {

    name "Test Process SRST2_RECN"
    script "modules/srst2_recN.nf"
    process "SRST2_RECN"

    test("Should map reads to recN reference") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = ["test", file("${projectDir}/tests/data/input_S999_R1_001.fastq.gz"), file("${projectDir}/tests/data/input_S999_R2_001.fastq.gz")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
