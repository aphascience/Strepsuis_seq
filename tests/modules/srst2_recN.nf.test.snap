{
    "Should map reads to recN reference": {
        "content": [
            {
                "0": [
                    [
                        "test",
                        "recN_test__genes__Ss-recN__results.txt:md5,0bb089d752d10b85412a8e61164d8bfa"
                    ]
                ],
                "1": [
                    [
                        "test",
                        "recN_test__fullgenes__Ss-recN__results.txt:md5,16f7fbd03d482f3a53988763d44c5276"
                    ]
                ],
                "recN_fullgenes": [
                    [
                        "test",
                        "recN_test__fullgenes__Ss-recN__results.txt:md5,16f7fbd03d482f3a53988763d44c5276"
                    ]
                ],
                "recN_genes": [
                    [
                        "test",
                        "recN_test__genes__Ss-recN__results.txt:md5,0bb089d752d10b85412a8e61164d8bfa"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-24T10:14:20.086863523"
    }
}